modules/custom/openy_activity_finder/js/activity_finder.js

```diff
@@ -350,49 +428,179 @@
         component.updateSearchQuery();
-        component.checkFilters(1);
       });
       component.$watch('checkedDays', function(){
         component.updateSearchQuery();
-        component.checkFilters(1);
       });
       component.$watch('checkedProgramTypes', function(){
         component.updateSearchQuery();
-        component.checkFilters(1);
       });
       component.$watch('checkedCategories', function(){
         component.updateSearchQuery();
-        component.checkFilters(2);
       });
       component.$watch('checkedLocations', function(){
         component.updateSearchQuery();
-        component.checkFilters(3);
       });
       // Get url from paragraph's field.
       component.afResultsRef = 'OpenY' in window ? window.OpenY.field_prgf_af_results_ref[0]['url'] : '';
       // Get 1/0 from paragraph's field.
-      component.hideProgramStep = $('.field-prgf-hide-program-categ').text();
+      component.hideProgramStep = 'OpenY' in window ? window.OpenY.field_prgf_hide_program_categ[0]['value'] : '';
       // Get 1/0 from paragraph's field.
-      component.hideLocationStep = $('.field-prgf-hide-loc-select-step').text();
+      component.hideLocationStep = 'OpenY' in window ? window.OpenY.field_prgf_hide_loc_select_step[0]['value'] : '';
       if (this.hideProgramStep == 1) {
         this.total_steps--;
       }
       if (this.hideLocationStep == 1) {
         this.total_steps--;
       }
+
+      // We need to rebuild breadcrumbs because they depend on DOM structure
+      // and it is being rendered by javascript. Once it is rendered, we can
+      // run our jQuery queries to get labels from input elements.
+      var component = this;
+
+      // Instead of fixed time we check if locations checkboxes got displayed.
+      var checkExist = setInterval(function() {
+        if ($('#locations-accordion-group input').length) {
+          component.rebuildBreadcrumbs = 1;
+          clearInterval(checkExist);
+        }
+      }, 200);
+    },
+    watch: {
+      // Call again the method if the route changes.
+      '$route': function() {
+
+        // If for some reason steps start increasing infinitely it simply hangs
+        // the browser. So we have a little protection here.
+        if (this.step > 5) {
+          console.log(this.step, 'FATAL');
+          return;
+        }
+
+        var stepGet = decodeURIComponent(this.$route.query.step);
+
+        // @TODO Make transition between step smoother so we do not need
+        // to run function next() multiple times.
+        if (stepGet != this.step) {
+          this.step = 0;
+          var i;
+          for (i = 0; i < stepGet; i++) {
+            this.next(false);
+          }
+        }
+
+        if (typeof this.$route.query.ages != 'undefined') {
+          var checkedAgesGet = decodeURIComponent(this.$route.query.ages);
+          if (JSON.stringify(checkedAgesGet.split(',')) != JSON.stringify(this.checkedAges)) {
```

Comment:
shouldn't we sort the, before doing stringify?

Reply:
actually we use it only like "length" comparing, so no worries.